src/main/java/com/limechain/runtime/hostapi/OffchainHostFunctions.java

```diff
@@ -75,4 +101,109 @@ public static List<ImportObject> getFunctions() {
+     *
+     * @return an integer which is equal to 1 if the local node is a potential validator or an integer equal to 0 if
+     * it is not.
+     */
+    public int extOffchainIsValidator() {
+        return config.getNodeRole() == NodeRole.AUTHORING ? 1 : 0;
+    }
+
+    /**
+     * Given a SCALE encoded extrinsic, this function submits the extrinsic to the Host’s transaction pool,
+     * ready to be propagated to remote peers.
+     *
+     * @param extrinsicPointer a pointer-size to the byte array storing the encoded extrinsic.
+     * @return a  pointer-size to the SCALE encoded Result value. Neither on success nor failure is there any
+     * additional data provided. The cause of a failure is implementation specific.
+     */
+    public RuntimePointerSize extOffchainSubmitTransaction(RuntimePointerSize extrinsicPointer) {
+        byte[] extrinsic = hostApi.getDataFromMemory(extrinsicPointer);
+        // TODO: add to transaction pool,  when implemented, and set success to the result of that operation
+        boolean success = true;
+
+        return hostApi.writeDataToMemory(scaleEncodedEmptyResult(success));
+    }
+
+    private byte[] scaleEncodedEmptyResult(boolean success) {
+        Result.ResultMode resultMode = success ? Result.ResultMode.OK : Result.ResultMode.ERR;
+        return new byte[] { resultMode.getValue() };
+    }
+
+    /**
+     * Returns the SCALE encoded, opaque information about the local node’s network state.
+     *
+     * @return a pointer-size to the SCALE encoded Result value.
+     * On success - it contains the Opaque network state structure .
+     * On failure, an empty value is yielded where its cause is implementation specific.
+     *
+     * @see <a href=https://spec.polkadot.network/chap-host-api#defn-opaque-network-state>Opaque Network State</a>
+     */
+    public RuntimePointerSize extOffchainNetworkState() {
+        Network network = Network.getNetwork();
+        PeerId peerId = network.getHost().getPeerId();
+        List<Multiaddr> multiAddresses = network.getHost().listenAddresses();
+
+        return hostApi.writeDataToMemory(scaleEncodedOpaqueNetwork(peerId, multiAddresses));
+    }
+
+    private byte[] scaleEncodedOpaqueNetwork(PeerId peerId, List<Multiaddr> multiAddresses) {
+        try (ByteArrayOutputStream buf = new ByteArrayOutputStream();
+             ScaleCodecWriter writer = new ScaleCodecWriter(buf)) {
+            ByteBuffer data = ByteBuffer.wrap(peerId.getBytes());
+            multiAddresses.stream().map(Multiaddr::serialize).forEach(data::put);
+
+            Result<byte[], Exception> result = new Result<>(Result.ResultMode.OK, data.array(), null);
+
+            new ResultWriter<byte[], Exception>()
+                    .writeResult(writer, ScaleCodecWriter::writeByteArray, null, result);
+            return buf.toByteArray();
+        } catch (IOException e) {
+            return scaleEncodedEmptyResult(false);
+        }
```

Comment:
We always want to have access to the exception stack trace without having to debug the application. I will leave it up to you to determine the correct logging level, judging by the fact that we are not throwing an exception `error` would probably be too high of a level, but it's hard for me to determine whether `warning` is just right or too high.

Reply:
Added a `warning` log